util/types/time.go

```diff
@@ -965,7 +960,7 @@ func ParseDuration(str string, fsp int) (Duration, error) {
 	case 3:
 		// Time format maybe HH:MM:SS or HHH:MM:SS.
 		// See https://dev.mysql.com/doc/refman/5.7/en/time.html
-		if !dayExists && len(seps[0]) == 3 {
```

Comment:
because in MySQL, `select cast('-24 100:00:00' as time);` is valid. see https://dev.mysql.com/doc/refman/5.7/en/time.html and https://dev.mysql.com/doc/refman/5.7/en/date-and-time-literals.html @tiancaiamao